dev/src/Domain/Supplier.java

```diff
@@ -45,6 +46,11 @@ public List<ProductInContract> getAllProductsInContracts() {
                 .collect(Collectors.toList());
     }
 
+    public List<Contract> getContractsSupplying(String productName) {
```

Comment:
We want to enable for the same supplier multiple contracts with the same product? 
I think it's not a given requirement for the system.

Reply:
We can change it to return only the supplier Contract which has the product